### ISSUE TITLE
Filesize limitation on encrypted transfers

### ISSUE THREAD

**Issue by koshui** [filesender/filesender#816]
Trying to download a 3.1 GB file ends in "Decrypting: NaN %" at the end of the file transfer at the point where the decryption should start. My guess is that it is just too big.
Tested on Ubuntu 18.04 ,Windows 10 with Chrome.

**Comment by monkeyiq** [comment]
The current way FileSender decrypts data can create issues with running out of RAM. A very recent update in https://github.com/filesender/filesender/pull/817 will stream chunks of a file down from the server decrypting each as they arrive. I have tested this update on a 4gb file successfully but have not explored the limit of how large it should be able to now handle. This will be part of the next FileSender release.

**Comment by monkeyiq** [comment]
This should be much improved in the latest release https://github.com/filesender/filesender/releases/tag/master-filesender-2.19

**Comment by madsi1m** [comment]
Are you able to test with a 30gb file?

**Comment by monkeyiq** [comment]
I'll have a look at the aarnet mirros for something of around that size to test locally.

**Comment by monkeyiq** [comment]
I might end up just concatenating some of the 4gb iso distros into a 20gb and 40gb file

**Comment by monkeyiq** [comment]
I added the CentOS-6.10-x86_64-bin-DVD1.iso image (about 4gb) a bunch of times by using 'cat' to the end of file. I ended up with a 38gb file (testfile-40gb) which has an md5sum of f48ac52f2af1a621da5bab58dd84c5ca  

This took a while to transfer but worked and the RAM usage in Chrome seemed stable (using top and shift+m to sort my memory size and other tests).

**Comment by monkeyiq** [comment]
Of the many settings I had tested during development this above 38gb test was done using the following config.php fragment. (for other readers mode 3 is AES-GCM)
```
$config['encryption_key_version_new_files'] = 3;
$config['storage_type'] = 'filesystemChunked';
```

**Comment by madsi1m** [comment]
very nice, very nice indeed!

**Comment by Phaze-III** [comment]
> I might end up just concatenating some of the 4gb iso distros into a 20gb and 40gb file

Back in the days I used a simple script like below to generate 'large' test files. Just increase/adapt the numbers :) 

```sh
#!/bin/sh

# 2 GB (and 1 MB below), 4 GB (and 1 MB below), 5 GB

# Set device to read bytes from: zero random or urandom (non-blocking on
# some Linux systems). When using /dev/zero compression will be a
# significant factor.
types="zero random"

for type in ${types}
do
   for size in 2047 2048 4095 4096 5120
   do
      dd if=/dev/${type} of=${size}M-${type}.dat bs=1048576 count=${size}
   done
done
```

**Comment by monkeyiq** [comment]
This should be fixed in version 2.19 released on 2 Aug. Versions before 2.19 will have issues where the amount of RAM required to decrypt large files will impose a limit on how large an encrypted file can be handled by a downloading machine.
